src/drivers/imx/esai.c

```diff
@@ -4,37 +4,456 @@
 //
 // Author: Daniel Baluta <daniel.baluta@nxp.com>
 
+#include <sof/drivers/edma.h>
 #include <sof/drivers/esai.h>
 #include <sof/lib/dai.h>
 #include <sof/lib/dma.h>
 #include <ipc/dai.h>
 
+struct esai_pdata {
+	struct {
+		uint32_t ecr;
+		uint32_t tfcr;
+		uint32_t rfcr;
+		uint32_t saicr;
+		uint32_t tcr;
+		uint32_t tccr;
+		uint32_t rcr;
+		uint32_t rccr;
+		uint32_t tsma;
+		uint32_t tsmb;
+		uint32_t rsma;
+		uint32_t rsmb;
+		uint32_t prrc;
+		uint32_t pcrc;
+	} regs;
+	// TODO set the config here, prepare the registers and defer the
+	// actual updating of the HW registers to the start() and stop()
+	// triggers
+};
+
+static void esai_regs_dump(struct dai *dai)
+{
+	/* Read every readable ESAI register and print it */
```

Comment:
Lets remove this from the initial submission.

Reply:
Done.